src/main/java/dinkplugin/notifiers/data/XpNotificationData.java

```diff
@@ -0,0 +1,41 @@
+package dinkplugin.notifiers.data;
+
+import dinkplugin.message.Field;
+import lombok.EqualsAndHashCode;
+import lombok.Value;
+import net.runelite.api.Experience;
+import net.runelite.api.Skill;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.SortedSet;
+import java.util.TreeSet;
+
+@Value
+@EqualsAndHashCode(callSuper = false)
+public class XpNotificationData extends NotificationData {
+    Map<Skill, Integer> xpData;
+    Collection<Skill> milestoneAchieved;
+    int interval;
+    transient String totalXp;
+
+    @Override
+    public List<Field> getFields() {
+        List<Field> fields = new ArrayList<>(2);
+        fields.add(new Field("Total XP", Field.formatBlock("", totalXp)));
```

Comment:
do we care about a total XP field at all?

Reply:
I don't, probably remove until we get a bit more of a vibe for the feature